File: apps/android/app/src/main/java/dev/tohure/android_rust_test/ui/arithmetic/ArithmeticUiState.kt
package dev.tohure.android_rust_test.ui.arithmetic

import androidx.compose.runtime.Immutable

/** Las dos operaciones que expone el core. */
enum class Operation { ADD, SUBTRACT }

@Immutable
data class ArithmeticUiState(
    val operandA: String = "0.1",
    val operandB: String = "0.2",
    val operation: Operation = Operation.ADD,
    /** Lo que devuelve el core: `Decimal` con la escala correcta. */
    val coreResult: String = "",
    /** Lo que devuelve el flotante nativo. Existe para exhibir la divergencia. */
    val nativeResult: String = "",
    val error: String? = null,
)

File: apps/ios/ios-rust-test/UI/Arithmetic/ArithmeticUiState.swift
/// Las dos operaciones que expone el core.
enum Operation: CaseIterable {
    case add, subtract

    /// Label exacto de `docs/ui-spec.md`.
    var label: String {
        switch self {
        case .add: return "Sumar"
        case .subtract: return "Restar"
        }
    }
}

struct ArithmeticUiState {
    /// Arranca en el caso `ar-001` del contrato: es el que mejor exhibe la divergencia.
    var operandA = "0.1"
    var operandB = "0.2"
    var operation: Operation = .add
    /// Lo que devuelve el core: `Decimal` con la escala correcta. **String, siempre.**
    var coreResult = ""
    /// Lo que devuelve el flotante nativo. Existe para exhibir la divergencia.
    var nativeResult = ""
    /// Ya resuelto a texto de usuario, no la excepción.
    var error: String?
}
